src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs

```diff
@@ -198,6 +198,7 @@ internal static class IDEDiagnosticIds
     public const string UseCollectionExpressionForCreateDiagnosticId = "IDE0303";
     public const string UseCollectionExpressionForBuilderDiagnosticId = "IDE0304";
     public const string UseCollectionExpressionForFluentDiagnosticId = "IDE0305";
+    public const string ConvertToInterpolatedStringDiagnosticId = "IDE0310";
```

Comment:
These should be called `UseInterpolatedString`.  `Convert` is hte action to get yo uto the desired state.  But the diagnostic id states the desired style preference.  In this case, it's to use interpolated strings.

Reply:
Note; we will likely end up wanting separate diagnostic ids for each pattern we detect.  Stephen listed several.  this will make it possible for users to enable/disable the diagnostic for some cases htey don't wnat .

Reply:
Will keep this conversation open to remind myself about multiple patterns.